logstreams/src/main/java/io/zeebe/logstreams/log/BufferedLogStreamReader.java

```diff
@@ -109,12 +108,40 @@ public void seekToFirstEvent() {
     seek(FIRST_POSITION);
   }
 
-  @Override
-  public void seekToLastEvent() {
-    seek(getLastPosition());
+  public long seekToEnd() {
+    // invalidate events first as the buffer content may change
+    invalidateBufferAndOffsets();
 
-    if (isNextEventInitialized()) {
-      checkIfNextEventIsCommitted();
+    final long blockAddress = logStorage.getFirstBlockAddress();
+    if (blockAddress < 0) {
+      // no block found => empty log
+      state = IteratorState.EMPTY_LOG_STREAM;
+      return -1;
+    } else {
+      readLastBlockIntoBuffer();
+      return completeEventsInBlockProcessor.getLastReadEventPosition();
+    }
+  }
+
+  private void readLastBlockIntoBuffer() {
+    if (byteBuffer.remaining() < LogEntryDescriptor.HEADER_BLOCK_LENGTH) {
+      compactBuffer();
+    }
+
+    final long result = logStorage.readLastBlock(byteBuffer, completeEventsInBlockProcessor);
+
+    if (result == LogStorage.OP_RESULT_INSUFFICIENT_BUFFER_CAPACITY) {
+      // it was not possible to read the block in the existing buffer => expand buffer
+      long nextCapacity = 2L * (long) byteBuffer.capacity();
```

Comment:
This is related to my comment about the whole method but: this snippet can refactored into an 'expandBuffer' method since it's also in readBlock and it would serve as implicit documentation (making the comment unnecessary)